torch/_tensor.py

```diff
@@ -46,7 +46,9 @@ def _rebuild_from_type_v2(func, new_type, args, state):
     if new_type is Tensor:
         return func(*args)
 
-    ret = func(*args).as_subclass(new_type)
+    ret = func(*args)
+    if type(ret) is not new_type:
+        ret = ret.as_subclass(new_type)
```

Comment:
This pattern comes up a few times - does it make sense to subsume the `if type(x) is not subclass_type` guard into `as_subclass()` itself?

Reply:
This one sounds ok as in the case where the Tensor has a storage, we do use the usual Tensor serialization to save and load, then slap the subclass and its attributes onto it.
The other instances should go though. So I don't think this is necessary.